dashboard/app/models/programming_method.rb

```diff
@@ -74,4 +82,23 @@ def parsed_parameters
   def parsed_examples
     examples.blank? ? [] : JSON.parse(examples)
   end
+
+  def validate_overload
+    # No overload is always valid
+    return if overload_of.blank?
+    if overload_of == key
+      errors.add(:overload_of, "Cannot overload self")
+      return
+    end
+    if ProgrammingMethod.where(programming_class_id: programming_class_id, overload_of: key).count > 0
+      errors.add(:overload_of, "Cannot have overload if another method overloads it")
```

Comment:
nit: this error message is a little confusing. Maybe instead say 'This method cannot have an overload if another method overloads it'?